api_app/analyzers_manager/observable_analyzers/greynoise_labs.py

```diff
@@ -0,0 +1,287 @@
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {self._auth_token}",
+        }
+
+        try:
+            for key, value in queries.items():
+                if not value["ip_required"]:
+                    if not os.path.isfile(value["db_location"]) and not self.update():
+                        raise AnalyzerRunException(f"Failed extraction from {key} db")
+                    if not os.path.exists(value["db_location"]):
+                        raise AnalyzerRunException(
+                            f"database location {value['db_location']} does not exist"
+                        )
+
+                    with open(value["db_location"], "r", encoding="utf-8") as f:
+                        db = f.read()
+
+                    db_list = db.split("\n")
+                    if self.observable_name in db_list:
+                        result[key] = {"found": True}
+                    else:
+                        result[key] = {"found": False}
+                    continue
+
+                json_body = {
+                    "query": value["query_string"],
+                    "variables": {"ip": f"{self.observable_name}"},
+                }
+                response = requests.post(headers=headers, json=json_body, url=url)
+                response.raise_for_status()
+                result[key] = response.json()
+        except requests.RequestException as e:
+            raise AnalyzerRunException(e)
+
+        return result
+
+    @classmethod
+    def _get_auth_token(cls):
+        for plugin in PluginConfig.objects.filter(
+            parameter__python_module=cls.python_module,
+            parameter__is_secret=True,
+            parameter__name="auth_token",
+        ):
+            if plugin.value:
+                return plugin.value
+        return None
+
+    @classmethod
+    def _update_db(cls, auth_token: str):
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {auth_token}",
+        }
+
+        try:
+            logger.info("Fetching data from greynoise API (Greynoise_Labs).....")
+            response = requests.post(
+                headers=headers,
+                json={"query": queries["topc2s"]["query_string"]},
+                url=url,
+            )
+            response.raise_for_status()
+            topc2s_data = response.json()
+
+            with open(db_location, "w", encoding="utf-8") as f:
+                for value in topc2s_data["data"]["topC2s"]["c2s"]:
+                    ip = value["source_ip"]
+                    if ip:
+                        f.write(f"{ip}\n")
+
+            if not os.path.exists(db_location):
+                return False
+
+            logger.info("Data fetched from greynoise API (Greynoise_Labs).....")
+            return True
+        except Exception as e:
+            logger.exception(e)
+
+    @classmethod
+    def update(cls):
+        auth_token = cls._get_auth_token()
```

Comment:
Here. But how to get the auth_token during testing? Like which file should I look for?

Reply:
first we would need a new test for the update like we have for Feodo Tracker for instance : https://github.com/intelowlproject/IntelOwl/pull/2126/files

then, do not trigger Exception here:
```
                    if not os.path.isfile(value["db_location"]) and not self.update():
                        raise AnalyzerRunException(f"Failed extraction from {key} db")
```
in this way not only the test would work but the analyzer should not fail if 1 query out of 3 fails.
Please just write an error log but do not trigger the AnalyzerRunException that makes all the analyzer to fail